internal/cmd/freeze-api/README.md

```diff
@@ -0,0 +1,88 @@
+freeze-api
+==========
+
+Introduction
+------------
+The `freeze-api` tool is the official way to create public APIs for sensu-go.
+Public APIs are created from internal APIs. Internal APIs are housed in
+`github.com/sensu/sensu-go/internal/apis`. When the development team decides it
+is time for an internal API to be publicized, then a public, versioned API will
+be created in `github.com/sensu/sensu-go/apis`.
+
+What does the `freeze-api` tool actually do?
+--------------------------------------------
+The `freeze-api` tool has two key responsibilities.
+
+1. Create the versioned package hierarchy based on the internal package hierarchy.
+2. Generate conversion functions for the public API data types. This will allow
+the data types to be converted to internal data types.
+
+How can developers create a public sensu-go API?
+------------------------------------------------
+Follow these steps to create a public API for sensu-go.
+
+1. Create one or more datatypes that embed `meta.TypeMeta` and `meta.ObjectMeta`
+in the `internal/apis` package.
+2. Run the `go-to-protobuf` tool. This needs to be formalized somewhat, but see
+the `hack` directory for more information. This tool will generate a `.proto`
+file and compile it to Go source code.
+3. Implement any business logic required in the API.
+4. Run the `freeze-api` tool on the internal package, creating a new versioned
+package in `github.com/sensu/sensu-go/apis`. Use a kubernetes-style version
+name for the package. If you aren't sure what this means, have a look at the
+existing public APIs.
```

Comment:
Could we provide a concrete example here or in the help usage? I had to go through the code to figure out the parameters expected. E.g. `freeze-api -from github.com/sensu/sensu-go/internal/apis/rbac -to github.com/sensu/sensu-go/apis/rbac/v1alpha1`